client/landing/gutenboarding/hooks/use-typer.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useInterval } from 'lib/interval/use-interval';
+
+type TyperMode = 'TYPING' | 'DELETING';
+
+type AllTyperOptions = {
+	delayBetweenCharacters: number;
+	delayBetweenWords: number;
+};
+
+// if options are passed. At least one of them should be set
+type TyperOptions =
+	| Pick< AllTyperOptions, 'delayBetweenCharacters' >
+	| Pick< AllTyperOptions, 'delayBetweenWords' >
+	| AllTyperOptions;
+
+const defaultOptions: AllTyperOptions = { delayBetweenCharacters: 120, delayBetweenWords: 1200 };
+
+/**
+ * A React hook that returns typing-machine animated strings
+ *
+ * @param words An array of strings you want to create the typing effect for
+ * @param enabled Whether the animation is enabled
+ * @param options Animation options
+ * @param options.delayBetweenCharacters The typing delay time between two charactors. Default is 120ms
+ * @param options.delayBetweenWords The typing delay time between two words. Default is 1200ms
+ * @returns truncated strings from `words` array.
+ * The word get longer at every passing `delayBetweenCharacters` until it's fully spelled,
+ * then the next word is spelled out after `delayBetweenWords` passes.
+ */
+export default function useTyper(
+	words: Array< string >,
+	enabled = true,
+	options: TyperOptions = defaultOptions
```

Comment:
We'll be merging anyways, we don't need to set to the default object then merge it with itself:

```tsx
	options?: Options
```

Here, we'll use the `Partial` mapped type to make all properties optional.

```suggestion
	options?: Partial< Options >
```

Reply:
TIL: you can spread `undefined` 😮 

```js
{...undefined} == {}
```